applications/FluidDynamicsApplication/trilinos_extension/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+message("FOUND TRILINOS EXTENSIONS")
+
+include_directories(
+  ${CMAKE_SOURCE_DIR}/kratos
+  ${CMAKE_SOURCE_DIR}/applications/FluidDynamicsApplication
+  ${CMAKE_SOURCE_DIR}/applications/TrilinosApplication
+)
+
+if (NOT TRILINOS_FOUND)
+  find_package(TRILINOS REQUIRED)
```

Comment:
I added checks for `TRILINOS_FOUND` (which is set by `find_package(TRILINOS)`) both here and in the trilinos CMakeLists to try to search for trilinos only once. It is probably not the cleanest idea, so suggestions for improvement are welcome.

Reply:
I suggest to merge this and then integrate it in the global CMakeLists, similar to what we do with MPI (and also Boost afaik)